lib/matplotlib/spines.py

```diff
@@ -534,3 +537,114 @@ def set_color(self, c):
         """
         self.set_edgecolor(c)
         self.stale = True
+
+
+class SpinesProxy:
+    """
+    A proxy to broadcast ``set_*`` method calls to all contained `.Spines`.
+
+    The proxy cannot be used for any other operations on its members.
+
+    The supported methods are determined dynamically based on the contained
+    spines. If not all spines support a given method, it's executed only on
+    the subset of spines that support it.
+    """
+    def __init__(self, spine_dict):
+        self._spine_dict = spine_dict
+
+    def __getattr__(self, name):
+        broadcast_targets = [spine for spine in self._spine_dict.values()
+                             if hasattr(spine, name)]
+        if not name.startswith('set_') or not broadcast_targets:
+            raise AttributeError(
+                f"'SpinesProxy' object has no attribute '{name}'")
+
+        def x(_targets, _funcname, *args, **kwargs):
+            for spine in _targets:
+                getattr(spine, _funcname)(*args, **kwargs)
+        x = functools.partial(x, broadcast_targets, name)
+        x.__doc__ = broadcast_targets[0].__doc__
+        return x
+
+    def __dir__(self):
+        names = []
+        for spine in self._spine_dict.values():
+            names.extend(name
+                         for name in dir(spine) if name.startswith('set_'))
+        return list(sorted(set(names)))
+
+
+class Spines(MutableMapping):
+    r"""
+    The container of all `.Spine`\s in an Axes.
+
+    The interface is dict-like mapping names (e.g. 'left') to `.Spine` objects.
+    Additionally it implements some pandas.Series-like features like accessing
+    elements by attribute::
+
+        spines['top'].set_visible(False)
+        spines.top.set_visible(False)
+
+    Multiple spines can be addressed simultaneously by passing a list. This
+    will return a `SpinesProxy` that broadcasts all ``set_*`` calls to it's
```

Comment:
```suggestion
    will return a `SpinesProxy` that broadcasts all ``set_*`` calls to its
```
but also, it should be made clear that _only_ `set_*` is broadcast.